apps/jenkins/chaos/app_pod_failure/test.yml

```diff
@@ -68,46 +62,58 @@
 
         ## PRE-CHAOS APPLICATION LIVENESS CHECK
 
-        - name: Verify that the AUT (Application Under Test) is running 
-          shell: >
-            kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers
-            -o custom-columns=:status.phase
-          args:
-            executable: /bin/bash
-          register: app_status
-          until: "'Running' in app_status.stdout"
-          delay: 10
-          retries: 12
-
-        - name: Get application pod name 
-          shell: >
-            kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers
-            -o=custom-columns=NAME:".metadata.name"
-          args:
-            executable: /bin/bash
-          register: app_pod_name
+        - block:
+            - name: Verify that the AUT is running
+              include_tasks: /common/utils/check_deployment_status.yml
 
-        ## APPLICATION FAULT INJECTION 
+            - name: Get application pod name 
+              shell: >
+                kubectl get pods -n {{ app_ns }} -l {{ app_label }} --no-headers
+                -o=custom-columns=NAME:".metadata.name"
+              args:
+                executable: /bin/bash
+              register: app_pod_name
 
-        - include_tasks: /chaoslib/pumba/pod_failure_by_sigkill.yaml 
-          vars:
-            action: "killapp"
-            app_pod: "{{ app_pod_name.stdout }}"
-            namespace: "{{ app_ns }}"
+        ## APPLICATION FAULT INJECTION
+
+            - include_tasks: /chaoslib/pumba/pod_failure_by_sigkill.yaml
+              vars:
+                action: "killapp"
+                app_pod: "{{ app_pod_name.stdout }}"
+                namespace: "{{ app_ns }}"
+                label: "{{ app_label }}"
+
+          when: lookup('env','DEPLOY_TYPE') == 'deployment'
+
+        - block:
+
+            - name: Verify that the AUT is running
+              include_tasks: /common/utils/check_deployment_status.yml
+
+            - name: Get application pod name
```

Comment:
apart from finding application pod name, other two tasks seems to be same for both deployment and statefulset, so, they can be moved out of this block.